src/words/environment/WordsSay.java

```diff
@@ -1,22 +1,49 @@
 package words.environment;
 import java.util.LinkedList;
 
+import words.ast.AST;
+import words.ast.ASTValue;
+import words.ast.ASTValue.ValueType;
+import words.exceptions.WordsInvalidTypeException;
+import words.exceptions.WordsProgramException;
+import words.exceptions.WordsRuntimeException;
+
 public class WordsSay extends WordsAction {
-	private String message;
-	// TODO: Should also allow expression
-	
-	public WordsSay(String message) {
-		this.message = message;
+	private AST messageExpression;
+	private String messageValue;
+
+	public WordsSay(AST messageExpression) {
+		this.messageExpression = messageExpression;
+	}
+
+	public WordsSay(String messageValue) {
+		this.messageValue = messageValue;
 	}
-	
+
 	@Override
 	public boolean isExecutable() {
 		return true;
 	}
 
 	@Override
-	protected void doExecute(WordsObject object, WordsEnvironment environment) {
-		object.setMessage(message);
+	protected void doExecute(WordsObject object, WordsEnvironment environment) throws WordsProgramException {
+		if (messageExpression != null) {
```

Comment:
Per above, I think this should be changed that the only way to construct it is with an expression, in which case this check isn't necessary (but we could have an assertion for it).